src/main/java/emiya/task/Deadline.java

```diff
@@ -0,0 +1,39 @@
+package emiya.task;
+
+import java.time.LocalDateTime;
+
+import emiya.datehandler.DateHandler;
+import emiya.emiyaexception.InvalidDateException;
+import emiya.emiyaexception.WrongDateFormatException;
+
+public class Deadline extends Task {
+
+    private final String dateOfDeadline;
+    private LocalDateTime localDateTime = null;
+
+    public Deadline(boolean isCompleted, String nameOfTask, String dateOfDeadline)
+            throws WrongDateFormatException, InvalidDateException {
+        super(isCompleted, nameOfTask);
+        this.localDateTime = DateHandler.determineDateTime(dateOfDeadline);
+        this.dateOfDeadline = dateOfDeadline;
+    }
+
+    @Override
+    public String toString() {

```

Comment:
Maybe instead of having an if-else that returns almost the same thing, it might be shorter if you could find a way to place either a " "(space) or a "X" within the brackets depending on the boolean isCompleted and only having one return statement!